sdk/src/Services/S3/Custom/Transfer/Internal/MultipartDownloadCommand.cs

```diff
@@ -111,5 +114,89 @@ private FileDownloadConfiguration CreateConfiguration()
                 _request.FilePath
             );
         }
+
+        #region Event Firing Methods
+
+        /// <summary>
+        /// Fires the DownloadInitiatedEvent to notify subscribers that the download has started.
+        /// This event is fired exactly once at the beginning of the download operation.
+        /// </summary>
+        private void FireTransferInitiatedEvent()
+        {
+            var transferInitiatedEventArgs = new DownloadInitiatedEventArgs(_request, _request.FilePath);
+            _request.OnRaiseTransferInitiatedEvent(transferInitiatedEventArgs);
+        }
+
+        /// <summary>
+        /// Fires the DownloadCompletedEvent to notify subscribers that the download completed successfully.
+        /// This event is fired exactly once when all parts have been downloaded and assembled.
+        /// Downloads are complete, so transferred bytes equals total bytes.
+        /// </summary>
+        /// <param name="response">The unified TransferUtilityDownloadResponse containing S3 metadata</param>
+        /// <param name="totalBytes">The total number of bytes in the file</param>
+        private void FireTransferCompletedEvent(TransferUtilityDownloadResponse response, long totalBytes)
+        {
+            var transferCompletedEventArgs = new DownloadCompletedEventArgs(
+                _request,
+                response,
+                _request.FilePath,
+                totalBytes,
+                totalBytes);
+            _request.OnRaiseTransferCompletedEvent(transferCompletedEventArgs);
+        }
+
+        /// <summary>
+        /// Fires the DownloadFailedEvent to notify subscribers that the download failed.
+        /// This event is fired exactly once when an error occurs during the download.
+        /// Uses the last known transferred bytes from progress tracking.
+        /// </summary>
+        /// <param name="totalBytes">Total file size if known, otherwise -1</param>
+        private void FireTransferFailedEvent(long totalBytes = -1)
```

Comment:
I'm curious how this plays with the callback that @philasmar is creating for failed requests. We shouldn't have 2 events for the same thing.